sdk/core/Azure.Core/tests/DynamicData/DynamicJsonTests.cs

```diff
@@ -844,6 +844,100 @@ public void CanDifferentiateBetweenNullAndAbsent()
             Assert.Throws<KeyNotFoundException>(() => _ = json["bar"]);
             Assert.Throws<KeyNotFoundException>(() => { if (json["bar"] == null) {; } });
         }
+
+        [Test]
+        public void CanGetBinaryDataPropertyAsBytes()
+        {
+            dynamic json = BinaryData.FromString("""{ "data" : null }""").ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+            BinaryData data = BinaryData.FromString("a");
+            json.Data = data;
+            byte[] bytes = json.Data;
+            CollectionAssert.AreEqual(data.ToArray(), bytes);
+        }
+
+        [Test]
+        public void CanSetBinaryDataValueAnonymousType()
+        {
+            dynamic json = BinaryData.FromString("""{ "data" : null }""").ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+            var value = new
+            {
+                foo = 1
+            };
+
+            BinaryData serializedValue = BinaryData.FromObjectAsJson(value);
+
+            // existing property
+            json.Data = serializedValue;
+
+            // new property
+            json.NewData = serializedValue;
+
+            // To get the value back from BinaryData property, we must first obtain it as BinaryData.
+            BinaryData data = json.Data;
+            BinaryData newData = json.NewData;
+
+            // And now you can use dynamic data to read the values.
+            dynamic first = data.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+            dynamic second = newData.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+            Assert.AreEqual(1, (int)first.Foo);
+            Assert.AreEqual(1, (int)second.Foo);
+        }
+
+        [Test]
+        public void CanSetBinaryDataValueModelType()
```

Comment:
@tg-msft, I believe this addresses your BYOM case, but let me know if you feel it doesn't.

If a user wanted to add the BYOM model in a way that they could dot-into the JSON properties immediately after (i.e. without a cast through BinaryData), they could assign it as a DynamicData or a JsonElement - I will add a test to show this.

Reply:
What's the value of `json.Data` below?  Can I use and manipulate `json.Data.Number`?  Or is `json.Data` a base64 encoded string?   The BYOM scenario needs `json.Data.Number`.

Reply:
You would need to cast json.Data to a BinaryData first and then call ToObjectFromJson or ToDynamicFromJson to get to the Number as a property.

I see this as symmetric - you serialized and assigned a BinaryData, why not retrieve a BinaryData and deserialize?